docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst

```diff
@@ -1811,66 +1810,86 @@ for valid upgrades, :ref:`here <upgrade-model-reference>`.
-   actions exercised on interfaces. In this situation there may be
-   many versions of a template that implement the interface being
-   exercised.
+On processing a command submission, there are scenarios where the Ledger API server needs
+to disambiguate `by-package-name template IDs` to `by-package-id template IDs` (see :ref:`dynamic-package-resolution-in-command-submission`).
+For this purpose, the **package preference** concept is introduced
+as the mapping from package-name to package ID for all known package-names on the participant.
 
-In these situations the following rules are followed to resolve the
-package-name to a package-id:
+The package preference is needed at each command submission time and is assembled from two sources:
 
--  By default, the participant resolves a package-name to the package-id
-   pertaining to the highest package version uploaded
+- The **default** package preference of the Ledger API to which the command is submitted.
+  This is the exhaustive mapping from package-name to package ID for all known package-names on the participant.
+  If multiple package IDs exist for a package-name, the package ID pertaining to the package with the highest
+  version is used.
 
--  The command submitter can override the above-mentioned default
-   participant resolution by pinning package-ids in the Command’s
-   :ref:`package_id_selection_preference <com.daml.ledger.api.v1.Commands.package_id_selection_preference>`.
-   More specifically, this field is a list of package-ids that must
-   be explicitly used when resolving package-name *ambiguities* in
-   either command template-id or interface resolution.
+- The Ledger API client **override** as the
+  :ref:`package_id_selection_preference <com.daml.ledger.api.v1.Commands.package_id_selection_preference>` in a command submission.
+  This is a package-name to package ID resolution list explicitly provided by the client to
+  override the default package preference mentioned above.
 
    - See :ref:`here <daml-script-package-preference>` for how to provide this in Daml-Script
 
-   -  **Note:** The Command’s
-      :ref:`package_id_selection_preference <com.daml.ledger.api.v1.Commands.package_id_selection_preference>`
+   -  **Note:** The :ref:`package_id_selection_preference <com.daml.ledger.api.v1.Commands.package_id_selection_preference>`
       must not lead to ambiguous resolutions for package-names,
       meaning that it must not contain two package-ids pointing to
       packages with the same package-name, as otherwise the submission will fail with
-      an ``INVALID_ARGUMENT`` error
+      an ``INVALID_ARGUMENT`` error.
+
+.. note::
+    **Important**: DAR uploads change the package preference on the participant.
+    If a new uploaded DAR contains an upgradable (LF >= 1.17) package with a higher version
+    than the existing package preference for the same package name,
+    the preference is updated to the new package ID.
```

Comment:
```suggestion
    the default preference map is updated to the new package ID.
```

Reply:
👍 But I would not add map since I didn't mention it before